### DISCUSSION TITLE
Multimodal reference mapping on subsets

### DISCUSSION THREAD

**Discussion by erosix** [satijalab/seurat#4465] [Q&A] [answered]
Hello, I have a dataset originating from PBMCs but where i am only interested in T cells. I am sure there are pretty much only T cells inside now because I selected only cells that have a TCR (from the VDJ kit of 10x genomics). I tried to map the cell populations using the new reference mapping and the PBMCs reference dataset.
https://satijalab.org/seurat/articles/multimodal_reference_mapping.html

However, after having run the commands in the vignettt, including SCTtransform of my dataset, my cells are mapped mostly in T cell compartments but many are also forced in other compartments like monocytes, B cells and NK cells, which cannot be. Any idea how to deal with this? 
Will the function try to find anchors in all cell types although not all cell types are present in the query dataset? 
I also tried to remove non-T cells from the reference:

```
Idents(reference)="celltype.l1"
reference_T=subset(reference, ident=c("CD4 T","CD8 T", "other T"))
```
and re-run the FindAnchors and Mapping but it seems that then it is not able to distinguish populations at all, they do not cluster anymore in the "ref.umap" and cells are all mixed.
I appreciate any hint on this.

**Comment by jaisonj708** [comment]
I would check your mismapped cells to insure your sample has not been contaminated with other immune cells. Perhaps you may have some doublets.

**Reply by erosix** [reply]
Yes, I think in the end I would keep only cells that map correctly, to be on the safe side. Thank you for the hint.

**Comment by rsatija** [comment, marked as answer]
Just to add a bit more context - Azimuth does support the mapping of single cell types (or subsets of cells) to a complex reference. We have successfully tested taking isolated populations of T cells and mapping them to the reference, and we only map to T cells.

If you are getting a result where most cells map to T cells but some others do map to other cell types, we would suggest isolating cells that map - for example - to monocytes, and asking what genes are differentially expressed in these cells. If there are any monocyte or non-T markers, that would be evidence of contamination.

It is quite common to see contaminating cells, either due to FACS contamination, or alternately, doublet cells (for example, monocyte/T cell doublets may still express a TCR). Hope that helps-

**Reply by erosix** [reply]
Thank you. Good to hear that it should work also for subsets only.
